### PR TITLE
Gently demystify the sociology of nifti maskers

### PR DESCRIPTION
**Pull request by banilo**

### CODE REVIEW

File: nilearn/input_data/nifti_masker.py [hunk 1]
```diff
@@ -14,7 +14,10 @@
 
 
 class NiftiMasker(BaseMasker, CacheMixin):
-    """Nifti data loader with preprocessing
+    """Helper for masking of Niimg-like objects. NiftiMasker is useful when
+    preprocessing (detrending, standardization, resampling, etc.) of in-mask
+    voxels is necessary. Use case: working with time series of resting-state
+    or task maps.
```
Comment (bcipolli) [inline nilearn/input_data/nifti_masker.py]: This is helpful!  Adding `intersect_masks` to related functions would be appreciated as well.

I also suggest adding to the docstring of `fit`, to reminds users that only `mask_args` and `mask_strategy` are used for fitting.

Lastly, I would find it helpful to specify (here or in `transform`) that:
* Parameters (detrend, standardize) are computed from the images passed to `transform` (i.e. not computed and stored from `fit`)
* The inverse of these parameters is not applied when `inverse_transform` is called.

Reason for those last two suggestions: as someone coming from computer vision, I believe the setup here differs in the two ways above from how we (at least in our lab) preprocess our datasets.  So my expectations may be different than someone who comes from a neuroimaging background.

**Comment by lesteve** [comment]
Remark about form rather than content: our docstring tend to follow the [numpy docstring](https://github.com/numpy/numpy/blob/master/doc/HOWTO_DOCUMENT.rst.txt#docstring-standard) convention.

In particular (actually this one is an excerpt from [PEP 257](https://www.python.org/dev/peps/pep-0257/)):
> Multi-line docstrings consist of a summary line just like a one-line docstring, followed by a blank line, followed by a more elaborate description. 

Also I am not sure about the "Helper" wording.

**Comment by banilo** [comment]
> In particular (actually this one is an excerpt from PEP 257)

So, keep the first line where it is and introduce a blank line after?

> Also I am not sure about the "Helper" wording.

What do you suggest - "utility" or "tool" or just "class"?

**Comment by lesteve** [comment]
>> In particular (actually this one is an excerpt from PEP 257)

> So, keep the first line where it is and introduce a blank line after?

Yes.

>> Also I am not sure about the "Helper" wording.

> What do you suggest - "utility" or "tool" or just "class"?

No strong opinion, "class" would be fine but feel free to be creative.

**Comment by banilo** [comment]
PEP-ized that!

**Comment by lesteve** [comment]
OK merging this one, thanks. @bcipolli feel free to open a new issue about your comments.

**Comment by GaelVaroquaux** [comment]
Thank you Danilo!

**Pull request merged**